### PR TITLE
Added Talisman of the Farmer (approved suggestion)

### PR DESCRIPTION
**Pull request by Sfiguz7**
## Description

From 1.16 fortune can be applied (and works) on hoes too. It only makes sense for a crops related counterpart of the Miner Talisman (chance to dupe ore drops) to exist in Slimefun to reflect this addition.
The addition was suggested on Discord and has been approved on August 8th, 2021 (suggestion `#1473`)
## Proposed changes

- Added a Talisman of the Farmer
- Added a research for the Talisman of the Farmer
- Added to the TalismanListener BlockBreakEvent to reflect the change
- Added a new SlimefunTag for the Talisman triggers (same as Crop Growth Accelerator minus the pumpkin and melon stems

## Related Issues (if applicable)

Suggestion `#1473` on the Discord (approved on August 8th, 2021)
## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java [hunk 1]
```diff
@@ -364,6 +365,40 @@ public void onBlockDropItems(BlockDropItemEvent e) {
                     }
                 }
             }
+
+            // We also want to double crops
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java:369]: comment could be clearer
```suggestion
            // Handle double drops for Farmer Talisman
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java [hunk 2]
```diff
@@ -364,6 +365,40 @@ public void onBlockDropItems(BlockDropItemEvent e) {
...
+
+                        // We do not want to dupe blocks
+                        if (!droppedItem.getType().isBlock()) {
+                            int amount = Math.max(1, (dropAmount * 2) - droppedItem.getAmount());
+                            e.getBlock().getWorld().dropItemNaturally(e.getBlock().getLocation(), new CustomItemStack(droppedItem, amount));
+                            doubledDrops = true;
+                        }
+                    }
+
+                    // Fixes #2077
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java:391]: comment is about miner talisman

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java [hunk 3]
```diff
@@ -364,6 +365,40 @@ public void onBlockDropItems(BlockDropItemEvent e) {
...
+                    if (doubledDrops) {
+                        Talisman talisman = SlimefunItems.TALISMAN_FARMER.getItem(Talisman.class);
+
+                        // Fixes #2818
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java:395]: comment is about miner talisman

**Comment by svr333** [comment]
you renmoved the comments on the miner talisman not on the farmer talisman

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java [hunk 4]
```diff
@@ -364,6 +365,38 @@ public void onBlockDropItems(BlockDropItemEvent e) {
...
+
+                    if (doubledDrops) {
+                        Talisman talisman = SlimefunItems.TALISMAN_FARMER.getItem(Talisman.class);
+
+                        if (talisman != null) {
+                            talisman.sendMessage(e.getPlayer());
+                        }
+                    }
+                }
+            }
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java:399]: This is all almost identical with the Miner Talisman code above.
To combat this redundancy, you could create a `doubleDrops` method with parameters here

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Took me long enough to get to this.
Looks simple but I feel like this is way too redundant

**Review by TheBusyBiscuit — APPROVED** [review]
That looks much more organized!

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java [hunk 5]
```diff
@@ -10,6 +10,8 @@
 import javax.annotation.Nonnull;
 import javax.annotation.ParametersAreNonnullByDefault;
 
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java:13]: unused import

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]

**Review by svr333 — APPROVED** [review]

**Pull request merged**